pandas/_libs/writers.pyx

```diff
@@ -23,7 +23,7 @@ ctypedef fused pandas_string:
 @cython.boundscheck(False)
 @cython.wraparound(False)
 def write_csv_rows(list data, ndarray data_index,
```

Comment:
Do we stand to gain anything by being more explicit about the type of the ndarrays here?

Reply:
I wasn't able to find a type for these ndarrays that passes the test suite. Should be okay as is.